buildSrc/src/main/java/com/synopsys/integration/detect/battery/UpdateBatteryJava.java

```diff
@@ -0,0 +1,89 @@
+ * under the License.
+ */
+package com.synopsys.integration.detect.battery;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import org.gradle.api.DefaultTask;
+import org.gradle.api.tasks.TaskAction;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+// Occasionally changes to BDIO or other integral systems require mass changes to the battery. This facilitates that work.
+public class UpdateBatteryJava extends DefaultTask {
+
+    @TaskAction
+    public void updateBattery() throws IOException {
+        File batteryBuild = new File("build/battery/");
+        //battery.listFiles()
+        File batteryResources = new File("src/test/resources/battery/");
+
+        File batteryReports = new File("build/reports/tests/testBattery/classes/");
+        for(File report : batteryReports.listFiles()) {
+            for (String line : Files.readAllLines(report.toPath())) {
+                if (line.contains("***BDIO BATTERY TEST|")) {
+                    String chunk = extractChunk(line);
+                    String[] pieces = chunk.split("\\|");
+                    String testName = pieces[0];
+                    String resourcePrefix = pieces[1];
+                    String bdioName = pieces[2];
+
+                    File testFolder = new File(batteryResources, resourcePrefix);
+                    File bdioFolder = new File(testFolder, "bdio");
+                    File bdioFile = new File(bdioFolder, bdioName);
+                    //println(bdioFile.toString() + ":" + bdioFile.exists())
+
+                    File actualTestFolder = new File(batteryBuild, testName);
+                    File actualBdioFolder = new File(actualTestFolder, "bdio");
+                    File actualBdioFile = new File(actualBdioFolder, bdioName);
+
+                    //println(actualBdioFile.toString() + ":" + actualBdioFile.exists())
+
+                    JsonArray json = JsonParser.parseString(new String(Files.readAllBytes(actualBdioFile.toPath()))).getAsJsonArray();
+                    JsonObject headerElement = json.get(0).getAsJsonObject();
+                    headerElement.remove("@id");
+                    headerElement.remove("creationInfo");
+                    Gson gson = new GsonBuilder().setDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSX").setPrettyPrinting().create();
+                    FileWriter fileWriter = new FileWriter(bdioFile);
+                    fileWriter.write(gson.toJson(json));
+
+                    System.out.println("COPIED FROM: $actualBdioFile");
+                    System.out.println("COPIED TO  : $bdioFile");
+                }
+            }
+        }
+    }
+
+    private String extractChunk(String line) {
+        String chunk = line.split("\\*\\*\\*BDIO BATTERY TEST\\|")[1];
```

Comment:
Why the escaping here? Aren't we looking for *** and not \*\*\* or am I misinterpreting the escaping?

Reply:
If we are just trying to escape for regex, could we use Pattern.quote() instead?